common/static/coffee/src/discussion/utils.coffee

```diff
@@ -166,7 +166,14 @@ class @DiscussionUtil
       safeAjaxParams.error = => @discussionAlert(gettext("Sorry"), errorMsg)
     undo = _.pick(model.attributes, _.keys(updates))
     model.set(updates)
-    @safeAjax(safeAjaxParams).fail(() -> model.set(undo))
+    promise = @safeAjax(safeAjaxParams)
```

Comment:
@MatMoore just a heads up, if `safeAjax` would always return promise as I suggested above this change could be reverted.